CMakeLists.txt

```diff
@@ -39,6 +39,19 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
                "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif ()
 
+find_program(ccache_EXECUTABLE ccache)
+if(ccache_EXECUTABLE)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${ccache_EXECUTABLE})
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ${ccache_EXECUTABLE})
+    add_custom_target(ccache-stats
+        COMMAND ${ccache_EXECUTABLE} --show-stats
+    )
+else()
+    add_custom_target(ccache-stats
+        COMMAND echo "ccache not found."
+    )
+endif(ccache_EXECUTABLE)
```

Comment:
Please refer to vineyard or gae to see how to set the ccache flags. Using `RULE_LAUNCH_XXXX` may conflicit with some environment variables.